### ISSUE TITLE
Improve local module resolution

### ISSUE THREAD

**Issue by javier-garcia-meteologica** [neoclide/coc-eslint#56]
It's recommended that eslint installations are local to a project, that is, that eslint and all it's plugins are installed dependencies of the project. This is even more important after the release of eslint 7.0.

Right now, it's very difficult for coc-eslint to find local packages, let me explain why.

The [readme](https://github.com/neoclide/coc-eslint/tree/eed42192dab175bbf249e21c18c85cbd5afdd92a) shows that if the local package cannot be found, the settings `nodePath` or `packageManager` can be set to resolve packages globally. But there is no mention of settings for local packages.

And resolution of local packages is bound to fail. Coc-eslint uses the function [resolveSettings](https://github.com/neoclide/coc-eslint/blob/eed42192dab175bbf249e21c18c85cbd5afdd92a/server/index.ts#L294) to search for the eslint package.

Let's say the user is in a workspace, and he has not configured `nodePath` nor `packageManager`. The module will be searched like this:

```js
directory = settings.workspaceFolder ? URI.parse(settings.workspaceFolder.uri).fsPath : undefined
promise = resolveModule('./eslint', directory, nodePath).catch(() => {
  return resolveModule('eslint', directory, nodePath)
})
```

The variable `directory` will resolve to `/home/user/project`. The function call `resolveModule('./eslint', '/home/user/project', undefined)` will try to read the non-existing directory '/home/user/project/eslint', which will fail.

A workaround is to use `"eslint.nodePath": "./node_modules"`, but this is not intuitive and not documented.

### Proposed alternatives

1. Try to resolve first `./node_modules/eslint`. I don't know any project that doesn't use `<project_path>/node_modules` for storing modules.

```
promise = resolveModule('./node_modules/eslint', directory, nodePath).catch(() => {
  return resolveModule('./eslint', directory, nodePath).catch(() => {
    return resolveModule('eslint', directory, nodePath)
  });
});
```

2. Append the configuration variable `"eslint.localNodePath"`, which should use `node_modules` as the default value.

```js
      let directory: string
      if (uri.scheme === 'file') {
        directory = path.dirname(uri.fsPath)
      } else {
        directory = settings.workspaceFolder ? URI.parse(settings.workspaceFolder.uri).fsPath : undefined
      }
      if (directory && settings.localNodePath) {
        directory = path.resolve(directory, settings.localNodePath);
      }
```

**Comment by AdrienLemaire** [comment]
@javier-garcia-meteologica thank you very much, I was struggling to figure out why coc-eslint was using my system eslint (v7.3.0, currently broken with import/no-cycle) instead of the local one (v7.2, works fine), and adding `"eslint.nodePath": "./node_modules",` did the trick :heart:

**Comment by BenoitZugmeyer** [comment]
Sorry to kick in like this, but the [proposed fix](https://github.com/neoclide/coc-eslint/commit/0a658803c0dcb07e61a6609c2df0cb23bec12637) for this issue is the cause of the issue https://github.com/neoclide/coc-eslint/issues/77.

With all due respect, I'm not sure to understand the issue (and the commit supposed to fix it).

@javier-garcia-meteologica 
> The variable directory will resolve to /home/user/project. The function call resolveModule('./eslint', '/home/user/project', undefined) will try to read the non-existing directory '/home/user/project/eslint', which will fail.

This is absolutely true, but then the `.catch` will be invoked, and `resolveModule('eslint', directory, nodePath)` should be used as a fallback. This fallback will be in charge of resolving the local or global eslint module to use.

I tried to reproduce the issue you are facing. But with the initial condition you are mentioning (empty configuration, local setup of eslint in /project/node_modules/eslint), `coc-eslint` (before the proposed fix) is successfully resolving to the local eslint install (as shown in `:CocInfo`). Did I miss something?

**Comment by javier-garcia-meteologica** [comment]
If I use the code

```
      promise = resolveModule('./eslint', directory, nodePath).catch(() => {
        return resolveModule('eslint', directory, nodePath)
      })
```

Then it resolves to `/home/user/.config/yarn/global/node_modules/eslint/lib/api.js` instead of `<project>/node_modules/eslint`

**Comment by BenoitZugmeyer** [comment]
Thanks, I understand the issue now! I'll think of something.

**Comment by javier-garcia-meteologica** [comment]
I noticed that reversing the priority order makes module resolution work as expected . It picks up `<project>/node_modules/eslint` first.

```
      promise = resolveModule('eslint', directory, nodePath).catch(() => {
        return resolveModule('./eslint', directory, nodePath)
      })
```

@BenoitZugmeyer, could this fix #77?
